package.json

```diff
@@ -3,7 +3,10 @@
   "version": "1.6.2",
   "description": "Script to catch uncommitted/unversioned files, for CI",
   "bin": {
-    "catch-uncommitted": "catch-uncommitted.sh"
+    "catch-uncommitted": "catch-uncommitted.js"
+  },
+  "engines": {
+    "node": ">=8.0.0"
   },
   "scripts": {
     "test": "./catch-uncommitted.sh --skip-node-versionbot-changes"
```

Comment:
```suggestion
    "test": "node ./catch-uncommitted.js --skip-node-versionbot-changes"
```
Maybe.

Reply:
Good shout! I've amended slightly so as not to explicitly use `node` though, such that that test also verifies the `#!/usr/bin/env node`, and that the correct file execution permissions have been set.

Reply:
This is fine, as `catch-uncommitted` was already assuming a Unix environment by relying on a bash script. Windows cmd.exe / PowerShell (not WSL / Cygwin / MSYS...) does not understand [shebangs](https://en.wikipedia.org/wiki/Shebang_(Unix)) (at least the last time I checked) or file execution permissions, and would require `node ./catch-uncommitted.js`. Which brings me to the following point:

> Alternatively, to be able to support Zero Installs as well, I could port the shell script to JavaScript if you are open to the idea? Something along these lines so we don't have to introduce any external dependencies maybe?: [...]

Personally I didn't even know what [Zero Installs](https://yarnpkg.com/features/zero-installs) was about (I just googled it), :-) but I am always in favour of replacing shell script code with code in any semi-decent programming language, in this case JavaScript. And, by doing so, this PR would also be adding support for Windows (as long as the test script started with `node ...`). :-)  So yes, if you are still up to it, I would encourage replacing the bash script with a JS script like you suggested. A note regarding this line:

```
    execSync('hash git', { stdio: 'ignore' }).length !== 0
```

`hash git` is a bash/shell thing, but I understand that the only purpose of that line is to find out if `git` is installed / available in the PATH env var. So I think it could be replaced with `execSync('git', ...)` -- basically we would want `execSync` to fail if it cannot execute `git`.

Regarding:
```
  "engines": {
    "node": ">=8.0.0"
```

This PR being semver-major, as far as I am aware / concerned we could require `"node": ">=10.0.0"`, which would allow you to rely on ES2018 features and extended availability of the promises interface, e.g. `let { promises: fs } = require('fs')` (if you wanted).

If you prefer to just get this PR merged as it is (if you no longer have time to do the JS porting), that's fine too, 👍  just let us know.